app/models/target.rb

```diff
@@ -31,4 +33,15 @@ class Target < ApplicationRecord
                        numericality: { greater_than_or_equal_to: -90, less_than_or_equal_to: 90 }
   validates :longitude, presence: true,
                         numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
+
+  validate :target_limit_per_user, on: :create
```

Comment:
Maybe I am being too prickly, but... why only `on create`?

Reply:
Yeah this should be just a normal validation. Speaking of validations, I'm going to be a prick too and say that I don't like method validations. I always prefer [Validators](https://guides.rubyonrails.org/active_record_validations.html#custom-validators), even for trivial cases like these.